engines/docker/image_cache.go

```diff
@@ -0,0 +1,108 @@
+package dockerengine
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	docker "github.com/fsouza/go-dockerclient"
+	"github.com/taskcluster/taskcluster-worker/runtime"
+	"github.com/taskcluster/taskcluster-worker/runtime/atomics"
+	"github.com/taskcluster/taskcluster-worker/runtime/caching"
+	"github.com/taskcluster/taskcluster-worker/runtime/gc"
+)
+
+const dockerPullImageInactivityTimeout = 30 * time.Second
```

Comment:
please give this a doc comment explaining what it does... I assume it has to do with docker client assuming docker is failing if the response locks up...